src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
 export interface Figure {
+  shape: string;
+  color: string;
 
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: string = 'triangle';
 
+  color: string;
+
+  private readonly sides: number[];
+
+  constructor(color: string, a: number, b: number, c: number) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('All sides must be greater than 0');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(
+        'Invalid triangle: the sum of two'
+        + ' sides must be greater than the third side',
+      );
+    }
+
+    this.color = color;
+    this.sides = [a, b, c];
+  }
+
+  getArea(): number {
+    const [a, b, c] = this.sides;
+    const s = (a + b + c) / 2;
+
+    return +(Math.sqrt(s * (s - a) * (s - b) * (s - c))).toFixed(2);
+  }
 }
 
 export class Circle implements Figure {
+  shape: string = 'circle';
+
+  color: string;
+
+  radius: number;
 
+  constructor(color: string, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Radius must be greater than 0');
+    }
+
+    this.color = color;
```

Comment:
you can simplify it using the Typescript Constructor Shorthand


https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd